include/examples/simple_behavior.h

```diff
@@ -0,0 +1,192 @@
+   *          initialization that did not take place in the constructor should
+   *          occur, e.g., setting up pointers. In this example, the desired
+   *          torques are initialized to zero.
+   */
+  void Init() override
+  {
+    // Initializes desired torques (for user safety, these are set to zero)
+    desired_torques_ = std::vector<float>(robot_->joints.size(), 0);
+
+    // Indicate that behavior has been initialized
+    set_initialized();
+    std::fprintf(stdout, "Initialized %s\n", get_name().c_str());
+  }
+
+  /**
+   * @brief Starts the behavior
+   * @details This code runs a single time when the behavior starts up. This is
+   *          a good place to read in the initial robot state and set the
+   *          appropriate variables to begin the behavior. The previously active
+   *          behavior is passed as an argument, permitting different
+   *          functionalities for different prior behaviors. In this example,
+   *          the behavior is marked as active and the initial robot state is
+   *          read.
+   * @param prev_behavior the previously active behavior
+   */
+  void Begin(std::unique_ptr<kodlab::Behavior<SimpleRobot>> &prev_behavior) override
+  {
+    // Update internal state
+    state_ = RUNNING;
+
+    // Read starting joint positions
+    joint_start_positions_ = robot_->GetJointPositions();
+    std::fprintf(stdout, "%s has begun\n", get_name().c_str());
+  }
+
+  /**
+   * @brief Updates the behavior
+   * @details This code runs after Begin, and is called on every control loop
+   *          update. Use this to define the primary content of your behavior,
+   *          e.g., to update and command new joint torques.  Additionally, this
+   *          code can monitor if the behavior has terminated and update flags
+   *          appropriately. In this example, this function reads the updated
+   *          joint velocities and maintains state machine corresponding to the
+   *          member enum `state_`.
+   */
+  void Update() override
+  {
+    // Update joint velocities from robot
+    joint_velocities_ = robot_->GetJointVelocities();
+
+    switch (state_)
+    {
+      case RUNNING:  // in a running state
+        // Set joint torques to running levels
+        desired_torques_ =
+            std::vector<float>(robot_->joints.size(), RUN_TORQUE);
+        break;
+
+      case STOPPING:  // in a stopping state
+        // Lower the joint torques to slowing levels
+        desired_torques_ =
+            std::vector<float>(robot_->joints.size(), SLOW_TORQUE);
+
+        // Set behavior to inactive if average velocity is below a threshold
+        float avg_velocity =
+            std::accumulate(joint_velocities_.begin(),
+                            joint_velocities_.end(),
+                            0.0) / static_cast<float>(joint_velocities_.size());
+        if (avg_velocity <= 0.1)
+        {
+          set_inactive();
```

Comment:
The behavior should not set itself inactive, the behavior manager should do that.

Reply:
Removed in c9b815e2ebf6e107820b8e28ad824ffe0d777fbe.